consensus/polybft/stake_manager.go

```diff
@@ -60,37 +60,37 @@ var _ StakeManager = (*stakeManager)(nil)
 // stakeManager saves transfer events that happened in each block
 // and calculates updated validator set based on changed stake
 type stakeManager struct {
-	logger                  hclog.Logger
-	state                   *State
-	rootChainRelayer        txrelayer.TxRelayer
-	key                     ethgo.Key
-	supernetManagerContract types.Address
-	validatorSetContract    types.Address
-	maxValidatorSetSize     int
-	polybftBackend          polybftBackend
+	logger                       hclog.Logger
+	state                        *State
+	key                          ethgo.Key
+	stakeManagerContractAddr     types.Address
+	validatorSetContract         types.Address
+	maxValidatorSetSize          int
+	polybftBackend               polybftBackend
+	stakeManagerContractContract *contract.Contract
+	blockchain                   blockchainBackend
 }
 
 // newStakeManager returns a new instance of stake manager
 func newStakeManager(
 	logger hclog.Logger,
 	state *State,
-	rootchainRelayer txrelayer.TxRelayer,
 	key ethgo.Key,
-	validatorSetAddr, supernetManagerAddr types.Address,
+	validatorSetAddr, stakeManagerAddr types.Address,
 	blockchain blockchainBackend,
 	polybftBackend polybftBackend,
 	maxValidatorSetSize int,
-	dbTx *bolt.Tx,
+	dbTx *bolt.Tx, blockchainB blockchainBackend,
```

Comment:
You already have `blockchainBackend`

```suggestion
	blockchain blockchainBackend,
	polybftBackend polybftBackend,
	maxValidatorSetSize int,
	dbTx *bolt.Tx,
```